deps/mbedtls.mk

```diff
@@ -0,0 +1,74 @@
+## mbedtls
+
+ifeq ($(USE_GPL_LIBS), 1)
+	MBEDTLS_SRC = mbedtls-$(MBEDTLS_VER)-gpl
+else
+	MBEDTLS_SRC = mbedtls-$(MBEDTLS_VER)-apache
+endif
+MBEDTLS_URL = https://tls.mbed.org/download/$(MBEDTLS_SRC).tgz
+
+MBEDTLS_OBJ_SOURCE := $(BUILDDIR)/mbedtls-$(MBEDTLS_VER)/library/libmbedcrypto.$(SHLIB_EXT)
+MBEDTLS_OBJ_TARGET := $(build_shlibdir)/libmbedcrypto.$(SHLIB_EXT)
+
+MBEDTLS_OPTS := $(CMAKE_COMMON) -DUSE_SHARED_MBEDTLS_LIBRARY=ON \
+		-DENABLE_PROGRAMS=OFF -DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_INSTALL_RPATH=$(build_prefix) -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE
+
+ifeq ($(OS),WINNT)
+MBEDTLS_OPTS += -DENABLE_ZLIB_SUPPORT=OFF
+ifeq ($(BUILD_OS),WINNT)
+MBEDTLS_OPTS += -G"MSYS Makefiles" -DENABLE_TESTING=OFF
+endif
+else
+MBEDTLS_OPTS += -DENABLE_ZLIB_SUPPORT=ON
```

Comment:
also cc @malmaud do you know whether this makes a difference for MbedTLS.jl's usage of the library?